### ISSUE TITLE
Is the recall computed correctly?

### ISSUE THREAD

**Issue by RichardSieg** [experiencor/basic-yolo-keras#160]
First of all: thanks for your great package! It's really neat!
I was wondering about your computation of the recall in the loss. And it seems like the number of seen examples is always increased, no matter the epoch. In my model, one epoch goes through the training set exactly once. So the number of seen examples should be reset to 0 with the beginning of every epoch. Could you maybe elaborate on the computation of the recall? In my case it's decreasing the whole time and I wondered whether it's just that I used wrong parameters or it's not computed correctly (at least in my setting).

**Comment by experiencor** [comment]
Hi @RichardSieg! The total_recall is computed from the first example ever. I also agree that it makes more sense to compute it epoch-wise. And current_recall is computed for the current example only and is not affected by ```seen```.

If current_recall > total_recall, current_recall will pull total_recall up. In your case, current_recall is approaching 0 and does not improve at all. It drags total_recall to 0 too. Meaning that your network is performing worse and worse. This behavior is expected during warm-up stage, not in the actual training stage (refer to the readme for more information). 

Does it clear your doubts?

**Comment by RichardSieg** [comment]
Yep! Thank you. My model learned to predict no boxes at all, which resulted in a low loss, but close to 0 recall. I adjusted the parameters to fix it, in particular I set `object_scale=5.0, no_object_scale=0.5, coord_scale=5.0, class_scale=1.0`. We have a single class problem, so class detection is not the issue. Also I computed default anchor boxes using the generator and the model performed way worse with that. Also SSD and RetinaNet performed worse. Will look over the computation again. But is it correct that the boxes are returned like `[width box1, height box1, width box2, height box2...]` ?

**Comment by RichardSieg** [comment]
Also there is one difference in your implementation and the paper: In your implementation, the square root is not taken of width and height. According to the paper the wh loss should be
`loss_wh    = tf.reduce_sum(tf.square(tf.sqrt(true_box_wh)-tf.sqrt(pred_box_wh))     * coord_mask) / (nb_coord_box + 1e-6) / 2.`

**Comment by experiencor** [comment]
@RichardSieg You may refer to the ```decode_netout``` function in frontend.py for the structure of the output. 

Yes, my implementation is slightly differently from the original one at this point.
